tests/test_fast_scan.py

```diff
@@ -33,6 +33,38 @@ def test_PQ4_accuracy(self):
         assert recall_at_1 > 0.6
         # print(f'recall@1 = {recall_at_1:.3f}')
 
+    def test_PQ4_speed(self):
+        # only test while building with avx2
+        print(faiss.has_AVX2)
+        if not faiss.has_AVX2:
+            return
+
+        ds  = datasets.SyntheticDataset(32, 2000, 5000, 1000)
+        xt = ds.get_train()
+        xb = ds.get_database()
+        xq = ds.get_queries()
+
+        index = faiss.index_factory(32, 'PQ16x4')
+        index.train(xt)
+        index.add(xb)
+
+        t0 = time.time()
+        D1, I1 = index.search(xq, 10)
+        t1 = time.time()
+        pq_t = t1 - t0
+        print('PQ16x4 search time:', pq_t)
+
+        index2 = faiss.index_factory(32, 'PQ16x4fs')
+        index2.train(xt)
+        index2.add(xb)
+
+        t0 = time.time()
+        D2, I2 = index2.search(xq, 10)
+        t1 = time.time()
+        pqfs_t = t1 - t0
+        print('PQ16x4fs search time:', pqfs_t)
+        self.assertLess(pqfs_t * 10, pq_t)
```

Comment:
It would raise an error here If avx2 did not properly work.

Reply:
I see what the intention is, I'm just worried that it might be a flaky test. But fine IMO to try and see how it goes for a while.

Reply:
@h-vetinari I agree it is tricky to do timing tests. However, we've had so many timing regressions that were discovered months down the road by users that I am eager to try something to avoid them.